### PR TITLE
Some small optimizations for the meson build

### PR DESCRIPTION
**Pull request by t-8ch**
Details are in each commits message.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -1194,8 +1194,8 @@ if features['egl-x11']
 endif
 
 plain_gl = get_option('plain-gl').require(
-    get_option('libmpv') and gl_allowed,
-    error_message: 'libmpv was not enabled!',
+    gl_allowed,
+    error_message: 'gl was not enabled!',
 )
 features += {'plain-gl': plain_gl.allowed()}
```
Comment (sfan5) [inline meson.build:1200]: I can't actually find anything that depends on plain-gl, can you?

File: libmpv/test.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <libmpv/client.h>
+
+int main(void)
+{
+    mpv_handle *handle = mpv_create();
+    if (!handle)
+        return 1;
+    mpv_destroy(handle);
+}
```
Comment (sfan5) [inline libmpv/test.c:26]: missing return 0 + calling mpv_initialize wouldn't hurt.

File: meson.build (hunk 1 above)
Comment (Dudemanguy) [reply meson.build:1200]: plain-gl is supposed to be for compiling libmpv without any platform-specific stuff. So in theory, you would disable everything else, but enable plain-gl and libmpv.

Comment (t-8ch) [reply meson.build:1200]: But nothing uses `HAVE_PLAIN_GL`, always `HAVE_GL`, so we can drop the plain-gl feature.

Comment (t-8ch) [reply meson.build:1200]: I went with `features += {'gl': plain_gl.allowed()}`

File: libmpv/test.c (hunk 2 above)
Comment (t-8ch) [reply libmpv/test.c:29]: Done

**Comment by t-8ch** [comment]
> Some small gripes I have with meson:

Should be fixed

File: meson.build [hunk 3]
```diff
@@ -390,7 +390,8 @@ add_project_link_arguments(link_flags, language: ['c', 'objc'])
 
 # osdep
 cocoa = dependency('appleframeworks', modules: ['Cocoa', 'IOKit', 'QuartzCore'],
-                   required: get_option('cocoa'))
+                   required: get_option('cocoa').require(build_machine.system() == 'darwin',
```
Comment (Dudemanguy) [inline meson.build:393]: There's already a `darwin` bool in the meson.build. You can just use that (ditto for below).

File: meson.build [hunk 4]
```diff
@@ -1683,6 +1684,9 @@ if get_option('libmpv')
     headers = ['libmpv/client.h', 'libmpv/render.h',
                'libmpv/render_gl.h', 'libmpv/stream_cb.h']
     install_headers(headers, subdir: 'mpv')
+
+    libmpv_test = executable('libmpv-test', 'libmpv/test.c', link_with: [libmpv])
+    test('libmpv', libmpv_test)
```
Comment (kasper93) [inline meson.build:1689]: If we now have tests in meson, would be nice to call `meson test` in all `ci/*` scripts where applicable.

File: meson.build [hunk 5]
```diff
@@ -1194,13 +1195,10 @@ if features['egl-x11']
...
-    get_option('libmpv') and gl_allowed,
-    error_message: 'libmpv was not enabled!',
+    gl_allowed,
+    error_message: 'gl was not enabled!',
 )
-features += {'plain-gl': plain_gl.allowed()}
-if features['plain-gl']
-    features += {'gl': true}
-endif
+features += {'gl': plain_gl.allowed()}
```
Comment (sfan5) [inline meson.build:1201]: if plain_gl is disabled this will disable gl entirely

File: meson.build [hunk 6]
```diff
@@ -1194,13 +1195,10 @@ if features['egl-x11']
 endif
 
 plain_gl = get_option('plain-gl').require(
-    get_option('libmpv') and gl_allowed,
-    error_message: 'libmpv was not enabled!',
+    gl_allowed,
```
Comment (sfan5) [inline meson.build:1198]: and IMO this should be kept, plain-gl has no useful effect for cplayer

File: meson.build (hunk 5 above)
Comment (t-8ch) [reply meson.build:1201]: Indeed, I'll drop this part

File: meson.build (hunk 6 above)
Comment (t-8ch) [reply meson.build:1198]: If cplayer and libmpv are built in one configuration, then cplayer will contain this code, too.

File: meson.build [hunk 7]
```diff
@@ -1445,106 +1446,108 @@ endif
...
-macos_sdk_version_py = find_program(join_paths(source_root, 'TOOLS',
-                                    'macos-sdk-version.py'))
-macos_sdk_info = ['', '0.0']
 if darwin
-    macos_sdk_info = run_command(macos_sdk_version_py, check: true).stdout().split(',')
-endif
+    macos_sdk_version_py = find_program(join_paths(source_root, 'TOOLS',
+                                        'macos-sdk-version.py'))
+    macos_sdk_info = ['', '0.0']
+    if darwin
```
Comment (Dudemanguy) [inline meson.build:1453]: Woops missed this, but you don't need this conditional anymore.

File: meson.build [hunk 8]
```diff
@@ -1445,106 +1446,105 @@ endif
...
+        swift_prog = find_program(run_command(xcrun, '-find', 'swift', check: true).stdout().strip())
+        swift_ver_string = run_command(swift_prog, '-version', check: true).stdout()
+        verRe = '''
+    #!/usr/bin/env python3
+    import re
+    import sys
+    verRe = re.compile("(?i)version\s?([\d.]+)")
+    swift_ver = verRe.search(sys.argv[1]).group(1)
+    sys.stdout.write(swift_ver)
+    '''
```
Comment (sfan5) [inline meson.build:1494]: this probably fails because of the indentation now but you could also directly shorten this to
```py
verRe = '''import sys,re;sys.stdout.write(re.search("(?i)version\s?([\d.]+)", sys.argv[1]).group(1))'''
```

File: meson.build [hunk 9]
```diff
@@ -1445,106 +1446,105 @@ endif
...
+    endif
 
-macos_10_11_features = get_option('macos-10-11-features').require(
-    macos_sdk_version.version_compare('>=10.11'),
-    error_message: 'a suitable macos sdk version could not be found!',
-)
+    macos_10_11_features = get_option('macos-10-11-features').require(
+        macos_sdk_version.version_compare('>=10.11'),
+        error_message: 'a suitable macos sdk version could not be found!',
+    )
```
[the remark this answers is not in the document]
Comment (t-8ch) [reply meson.build:1470]: Good point. I'll drop the darwin part of the PR. Somebody who knows how all the different macos development parts fit together should do this.

[the remark this answers is not in the document]
Comment (Dudemanguy) [reply meson.build]: >Somebody who knows how all the different macos development parts fit together should do this.

AKA me because I wrote it hah. I recall sfan5 complained to me about the sdk always being checked before and I never fixed it for some reason (hope I didn't give some lame excuse because it's not that hard).

**Comment by Dudemanguy** [comment]
`features += {'plain-gl': plain_gl.allowed()}` is still there

**Review by Dudemanguy — APPROVED** [review]
I'll follow up with the mac stuff afterwards.
